cpp/pagerank_module/algorithm_online/pagerank.cpp

```diff
@@ -0,0 +1,231 @@
+#include <memory>
+#include <unordered_map>
+
+#include <mg_graph.hpp>
+#include "pagerank.hpp"
+
+namespace pagerank_online_alg {
+namespace {
+PageRankData context;
+std::uint64_t global_R;
+double global_epsilon;
```

Comment:
Isn't this unsafe in multithreaded env?

Reply:
This is correct, and it is a big downside of these dynamic methods. Furthermore, when Memgraph is restarted, this is guaranteed to not work. We already discussed implementing a thread-safe environment for storing dynamic context, however, the cost of engineering would surpass the benefits of having it at this moment.

Reply:
I would write a comment at least, just to show that you're aware of those problems.

Reply:
Done!